components/widgets/fires/fires-ranked/selectors.js

```diff
@@ -38,23 +38,40 @@ export const getYears = createSelector([getLatestDate], latest => {
   return years;
 });
 
+export const getFilterWeeks = createSelector([getLatestDate, getOptionsSelected], (latest, options) => {
+  const period = options.weeks && options.weeks.value || 4;
+  const endWeek = moment(latest).isoWeek();
+  const startWeek = moment(latest).subtract(period, 'weeks').isoWeek();
+  return {startWeek, endWeek};
+});
+
 export const getStatsByAdmin = createSelector(
-  [getData, getYears, getAdm1, getLocation],
-  (data, years, adm1, location) => {
+  [getData, getYears, getAdm1, getLocation, getFilterWeeks],
+  (data, years, adm1, location, filterWeeks) => {
     if (isEmpty(data) || isEmpty(years)) {
       return null;
     }
     let matchKey = 'iso';
     if (location.value !== 'global') matchKey = adm1 ? 'adm2' : 'adm1';
     const alertsByAdm = groupBy(data, matchKey);
-
+    const {startWeek, endWeek} = filterWeeks;
     const filteredAlertsByAdmin = Object.entries(alertsByAdm).map(
       ([adm, adminAlerts]) => {
-        const countsArray = years.map(year => {
-          const filteredYear = adminAlerts.filter(el => el.year === year);
-          return filteredYear.length > 0 ? sumBy(filteredYear, 'count') : 0;
-        });
-
+        let countsArray = [];
+        if (startWeek < endWeek) {
+          countsArray = years.map(year => {
+            const filteredYear = adminAlerts.filter(el => el.year === year);
```

Comment:
everything below here is repeated so you could place the if else inside the `.filter`

Reply:
I'm avoiding having to do logic on every iteration of the map. If the period between start week and end week doesn't cross a year then theres no need to filter. In the case of country or global dashboard there will be thousands of elements in `adminAlerts` so I want to avoid filtering if it's not necessary so we have decent performance.